python/cudf/cudf/utils/performance_tracking.py

```diff
@@ -27,6 +30,9 @@ def _get_color_for_nvtx(name):
 
 def _performance_tracking(func, domain="cudf_python"):
     """Decorator for applying performance tracking (if enabled)."""
+    global dummy_annotate
+    if dummy_annotate.domain is nvtx.nvtx.dummy_domain:
```

Comment:
Question: Should we me guarding this with just nvtx check or `"memory_profiling"` option we have in `cudf`?